internal/core/src/common/Tracer.cpp

```diff
@@ -134,6 +134,28 @@ StartSpan(const std::string& name, const std::shared_ptr<trace::Span>& span) {
 }
 
 thread_local std::shared_ptr<trace::Span> local_span;
+
+std::string
+GetTraceID() {
+    // !!! The span is not sent by the calling context, but saved in the thread local now.
+    // So we cannot get the accurate trace id if the thread is switched to execute another task.
+    // Because we don't use folly thread pool to yield multi task,
+    // the trace id is almost accurate by now.
+    // It's safe to access local_span without mutex, because the span is not sent to other threads.
+    if (local_span == nullptr) {
+        return std::string();
+    }
+    auto ctx = local_span->GetContext();
+    auto trace_id = ctx.trace_id();
+    // The span is noop if the trace is off, 00000000... will be returned,
+    // so return "0" string directly to distinguish from the no local span.
+    if (!trace_id.IsValid()) {
+        return "0";
+    }
+    auto rep = trace_id.Id();
+    return BytesToHexStr(rep.data(), rep.size());
```

Comment:
will the trace str will be calculated each log call? can this be avoided?

Reply:
yes, save the context at root will fix it. 
But in our current implementation of segcore, LOG is low rate oepartion when working on INFO level at segcore, so make a pre-calculation will generate make-no-sense opeartion for most search opeartion, should not be a good-choice.
At DEBUG level, we don't care the performance of such a string building.

Reply:
Maybe a lazy+pre calc could be done, could be optimized in future lgtm now